src/presign/participant.rs

```diff
@@ -1185,3 +1185,178 @@ impl PresignKeyShareAndInfo {
         Ok((private, ret_publics))
     }
 }
+
+#[cfg(test)]
+pub(super) use test::presign_record_set_is_valid;
+
+#[cfg(test)]
+mod test {
+    use std::{collections::HashMap, iter::zip};
+
+    use k256::Scalar;
+    use libpaillier::unknown_order::BigNumber;
+    use rand::{CryptoRng, Rng, RngCore};
+    use tracing::debug;
+
+    use crate::{
+        auxinfo,
+        errors::Result,
+        keygen,
+        messages::{Message, MessageType, PresignMessageType},
+        participant::ProcessOutcome,
+        presign::{Input, PresignRecord},
+        utils::{self, testing::init_testing},
+        Identifier, ParticipantConfig, ParticipantIdentifier, ProtocolParticipant,
+    };
+
+    use super::{PresignParticipant, Status};
+
+    fn deliver_all(
+        messages: &[Message],
+        inboxes: &mut HashMap<ParticipantIdentifier, Vec<Message>>,
+    ) {
+        for message in messages {
```

Comment:
I think you copied this code from elsewhere  because I have seen it before, but these loops are silly. The inner loop iterates over the HashMap for every message.

We could simply use `message.to()` to look up the correct inbox. This code will also silently fail if `message.to()` does not exist in the inboxes.

Reply:
Silly indeed. I fixed everywhere, thanks for noticing.